apps/server-asset-sg/src/features/workgroups/workgroup.controller.ts

```diff
@@ -1,72 +1,61 @@
-import {
-  Body,
-  Controller,
-  Delete,
-  Get,
-  HttpCode,
-  HttpException,
-  HttpStatus,
-  Param,
-  ParseIntPipe,
-  Post,
-  Put,
-  ValidationPipe,
-} from '@nestjs/common';
-import { RequireRole } from '@/core/decorators/require-role.decorator';
-import { Role } from '@/features/users/user.model';
-import { Workgroup, WorkgroupDataBoundary, WorkgroupId } from '@/features/workgroups/workgroup.model';
+import { Controller, Delete, Get, HttpCode, HttpException, HttpStatus, Post, Put } from '@nestjs/common';
+import { Authorize } from '@/core/decorators/authorize.decorator';
+import { Authorized } from '@/core/decorators/authorized.decorator';
+import { Boundary } from '@/core/decorators/boundary.decorator';
+import { CurrentUser } from '@/core/decorators/current-user.decorator';
+import { UsePolicy } from '@/core/decorators/use-policy.decorator';
+import { UseRepo } from '@/core/decorators/use-repo.decorator';
+import { User } from '@/features/users/user.model';
+import { Workgroup, WorkgroupData, WorkgroupDataBoundary } from '@/features/workgroups/workgroup.model';
+import { WorkgroupPolicy } from '@/features/workgroups/workgroup.policy';
 import { WorkgroupRepo } from '@/features/workgroups/workgroup.repo';
 
 @Controller('/workgroups')
+@UsePolicy(WorkgroupPolicy)
+@UseRepo(WorkgroupRepo)
 export class WorkgroupController {
   constructor(private readonly workgroupRepo: WorkgroupRepo) {}
 
-  @Get('/:id')
-  async show(@Param('id', ParseIntPipe) id: WorkgroupId): Promise<Workgroup> {
-    const workGroup = await this.workgroupRepo.find(id);
-    if (workGroup === null) {
-      throw new HttpException('not found', 404);
-    }
-    return workGroup;
+  @Get('/')
+  @Authorize.User()
```

Comment:
Yes, this just means that all users can access the API path that reads the workgroups. The list is then filtered within the route so that the user only gets the assets that he has access to.

Reply:
Yeah I get that, but this still means that every user can get a list of all the Workgroups, right? Maybe they should not be able to see which wokrgroups exist (even though this information does not get them anywhere)

Reply:
@TIL-EBP is this cleared up or would you prefer to talk about this in more detail?